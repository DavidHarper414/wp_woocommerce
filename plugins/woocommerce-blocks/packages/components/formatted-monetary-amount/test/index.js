/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import FormattedMonetaryAmount from '../index';

describe( 'FormattedMonetaryAmount', () => {
	describe( 'separators', () => {
		test( 'should add the thousand separator', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value="156345"
					currency={ {
						code: 'EUR',
						symbol: '€',
						thousandSeparator: '.',
						decimalSeparator: ',',
						minorUnit: 2,
						prefix: '',
						suffix: ' €',
					} }
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		} );

		test( 'should not add thousand separator', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value="156345"
					currency={ {
						code: 'EUR',
						symbol: '€',
						decimalSeparator: ',',
						minorUnit: 2,
						prefix: '',
						suffix: ' €',
					} }
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		} );

		test( 'should remove the thousand separator when identical to the decimal one', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value="156345"
					currency={ {
						code: 'EUR',
						symbol: '€',
						thousandSeparator: ',',
						decimalSeparator: ',',
						minorUnit: 2,
						prefix: '',
						suffix: ' €',
					} }
				/>
			);
			expect( console ).toHaveWarned();
			expect( component.toJSON() ).toMatchSnapshot();
		} );
	} );
	describe( 'suffix/prefix', () => {
		test( 'should add the currency suffix', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value="15"
					currency={ {
						code: 'EUR',
						symbol: '€',
						thousandSeparator: '.',
						decimalSeparator: ',',
						minorUnit: 2,
						suffix: ' €',
					} }
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		} );

		test( 'should add the currency prefix', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value="15"
					currency={ {
						code: 'EUR',
						symbol: '€',
						thousandSeparator: '.',
						decimalSeparator: ',',
						minorUnit: 2,
						prefix: '€ ',
					} }
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		} );
	} );

	describe( 'supports different value types', () => {
		test( 'should support numbers', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value={ 15.0 }
					currency={ {
						code: 'EUR',
						symbol: '€',
						thousandSeparator: '.',
						decimalSeparator: ',',
						minorUnit: 0,
						suffix: ' €',
					} }
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		} );

		test( 'should support strings', () => {
			const component = TestRenderer.create(
				<FormattedMonetaryAmount
					value="15.0"
					currency={ {
						code: 'EUR',
						symbol: '€',
						thousandSeparator: '.',
						decimalSeparator: ',',
						minorUnit: 0,
						prefix: '€ ',
					} }
				/>
			);
			expect( component.toJSON() ).toMatchSnapshot();
		} );
	} );
} );
