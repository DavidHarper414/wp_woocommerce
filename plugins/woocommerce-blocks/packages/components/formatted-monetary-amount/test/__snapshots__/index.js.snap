// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedMonetaryAmount separators should add the thousand separator 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  1.563,45 €
</span>
`;

exports[`FormattedMonetaryAmount separators should not add thousand separator 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  1563,45 €
</span>
`;

exports[`FormattedMonetaryAmount separators should remove the thousand separator when identical to the decimal one 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  1563,45 €
</span>
`;

exports[`FormattedMonetaryAmount suffix/prefix should add the currency prefix 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  € 0,15
</span>
`;

exports[`FormattedMonetaryAmount suffix/prefix should add the currency suffix 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  0,15 €
</span>
`;

exports[`FormattedMonetaryAmount supports different value types should support numbers 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  15 €
</span>
`;

exports[`FormattedMonetaryAmount supports different value types should support strings 1`] = `
<span
  className="wc-block-formatted-money-amount wc-block-components-formatted-money-amount"
>
  € 15
</span>
`;
